core/src/test/java/apoc/trigger/TriggerTest.java

```diff
@@ -129,40 +129,59 @@ public void testTimeStampTrigger() throws Exception {
         });
     }
 
-    // TODO NC:Why is this failing?
-    @Ignore
     @Test
-    public void testTxId() throws Exception {
-        db.executeTransactionally("CALL apoc.trigger.add('txinfo','UNWIND $createdNodes AS n SET n.txId = $transactionId, n.txTime = $commitTime',{phase:'after'})");
+    public void testTxIdAfterAsync() throws Exception {
+        db.executeTransactionally("CALL apoc.trigger.add('txinfo','UNWIND $createdNodes AS n SET n.txId = $transactionId, n.txTime = $commitTime',{phase:'afterAsync'})");
         db.executeTransactionally("CREATE (f:Bar)");
-        TestUtil.testCall(db, "MATCH (f:Bar) RETURN f", (row) -> {
+        org.neo4j.test.assertion.Assert.assertEventually(() ->
+                        db.executeTransactionally("MATCH (n:Bar) RETURN n", Map.of(),
+                                result -> {
+                                    final Node node = result.<Node>columnAs("n").next();
+                                    return (long) node.getProperty("txId", -1L) > -1L
+                                            && (long) node.getProperty("txTime") > start;
```

Comment:
What's the difference between checking this and that the properties are not null (and have been assigned)?

Reply:
If I understand correctly, in this case I left the previous assertions on lines 154 and 155:
```
assertEquals (true, (Long) ((Node) row.get ("f")). getProperty ("txId")> -1L);
assertEquals (true, (Long) ((Node) row.get ("f")). getProperty ("txTime")> start);
```
I just changed the test from "after" to "afterAsync" (by adding `assertEventually` as well).

I think that verifying only the existence of `txId` and` txTime` is not enough, I could get wrong results because, for example, the `txTime` is lower than the `start` one, or am I wrong?

Reply:
But the node was created without properties in the first place, so if they are present, that should be enough to know the trigger was fired, isn't it?

Anyway, this is not a biggie